Project-Descriptions-and-Plans/LLVM-Toolchain/llvm-toolchain-pl.md

```diff
@@ -0,0 +1,314 @@
+<!--- What is the impact of doing/not doing this project on the OpenHW ecosystem. Why is OpenHW best suited to do this project -->
+
+A processor is only useful with a robust, up-to-date, proven compiler tool chain.  CORE-V, being a strict RISC-V derivative, can use a standard RISC-V compiler tool chain, but in these circumstances will not be able to take advantage of any of the added features of the CORE-V processors.  This project will provide a baseline tool chain allowing these features to be used (albeit without optimization) and simulated from the source code level.
+
+Furthermore, the licensing scheme of LLVM (Apache license) makes it particularly appealing to industrial players.  Because it does not require the disclosure of compiler source code, chip manufacturers and system integrators can better protect their differentiating Intellectual Property assets.  This licensing scheme also creates additional opportunities for software companies which can provide high added-value services in the area of compiler tuning for hardware/system-centric companies.
+
+In work done a few years ago, code generated for Embench tests by LLVM 11.0 was 4% quicker that GCC 10.2 code. This is an additional reason to offer LLVM support for CORE-V processors.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+
+The "original" compiler used for the development of RISC-V cores is GCC, licensed under the GNU General Public License, requiring vendors to make source code available.  LLVM offers a "permissive" licensing scheme which some customers find more attractive, since it does not require the source to be made available. As a result LLVM-based compiler chains appear to be at the basis (in whole or in part) of a number of proprietary compilers.
+
+The downside is that there is less pressure on companies to contribute back to LLVM development.  This project will give OpenHW Group members a choice over which approach to adopt (make their toolchain code available or not).
+
+### Related work
+
+The upstream LLVM tool chain project already supports standard RISC-V, including all ratified ISA extensions.  RISC-V support in LLVM is strongly compliant to the RISC-V specification, but relatively immature in comparison to older architectures as x86, Arm and MIPS.
+
+Moreover, Clang/LLVM follows closely new ISA extension discussions as it already provides a mechanism to support yet to be ratified extensions.
+
+Other toolchains available for RISC-V are:
+
+- the GNU tool chain for RISC-V, for which a CORE-V specific project is in progress within OpenHW Group;
+- IAR RISC-V compiler
+
+The GNU tool chain project within OpenHW Group mirrors this project in its suport for CORE-V archhitecture.
+
+An OpenHW Group architecture project devoted to the CVA6 (Ariane) family of cores has passed the project concept gate.  There is a clear synergy between this project and the CVA6 project:
```

Comment:
Actually passed Plan Approved. However, see note above, is CVA6 LLVM under this project heading @jeremybennett @jquevremont ?

Reply:
Updated.